src/protocolgame.cpp

```diff
@@ -988,6 +988,26 @@ void ProtocolGame::parseRotateItem(NetworkMessage& msg)
 	addGameTaskTimed(DISPATCHER_TASK_EXPIRATION, &Game::playerRotateItem, player->getID(), pos, stackpos, spriteId);
 }
 
+void ProtocolGame::parseRuleViolationReport(NetworkMessage &msg)
+{
+	RuleViolationType_t report_type = (RuleViolationType_t) msg.getByte();
+	RuleViolationReasons_t report_reason = (RuleViolationReasons_t) msg.getByte();
+	const std::string target_name = msg.getString();
+	std::string player_comment = msg.getString();
+	std::string translation = "";
```

Comment:
Yes, it does. In C++, declaring a variable default-initializes it. You could export to a free function:

```cpp
std::string translation(NetworkMessage& msg, uint8_t reportType)
{
    switch (reportType) {
        case REPORT_TYPE_NAME:
            return msg.getString();
        case REPORT_TYPE_STATEMENT:
            const std::string& translation = msg.getString();
            msg.get<uint32_t>();
            return translation;
        default:
            return {};
    }
}
```

That way you won't need to allocate the temporary and you allocate inplace instead. The `const std::string&` there does not allocate and passes through to the caller.

Also, use `const std::string&` when you don't mean to change the contents, it allows some semantic optimizations such as eliding copies and avoiding indirections.